app/src/main/java/com/android/bookswap/ui/profile/EditProfile.kt

```diff
@@ -0,0 +1,137 @@
+
+              OutlinedTextField(
+                  _lastName.value,
+                  {
+                    _lastName.value = it
+                    dataUser.lastName = _lastName.value
+                  },
+                  Modifier.testTag("lastnameTbx").fillMaxWidth().padding(8.dp, 4.dp),
+                  label = { Text("Lastname") },
+                  placeholder = { Text("Doe", Modifier, Color.Gray) },
+                  keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Text),
+                  singleLine = true)
+
+              OutlinedTextField(
+                  _email.value,
+                  {
+                    _email.value = it
+                    dataUser.email = _email.value
+                  },
+                  Modifier.testTag("emailTbx").fillMaxWidth().padding(8.dp, 4.dp),
+                  label = { Text("Email") },
+                  placeholder = { Text("John.Doe@example.com", Modifier, Color.Gray) },
+                  keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email),
+                  singleLine = true)
+
+              OutlinedTextField(
+                  _phone.value,
+                  {
+                    _phone.value = it
+                    dataUser.phoneNumber = _phone.value
+                  },
+                  Modifier.testTag("phoneTbx").fillMaxWidth().padding(8.dp, 4.dp),
+                  label = { Text("Phone") },
+                  placeholder = { Text("+4122345678", Modifier, Color.Gray) },
+                  keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Phone),
+                  singleLine = true)
+
+              Row(Modifier.fillMaxWidth().padding(8.dp), Arrangement.SpaceEvenly) {
+                Button(
+                    {
+                      Log.d(
+                          "EditProfile_ClickBtn",
+                          "Save Clicked, User info: ${dataUser.printFull1Line()}")
+                      onSave(dataUser)
+                    },
+                    Modifier.testTag("confirmBtn")) {
+                      Text("Save")
+                    }
+
+                Button(
+                    { Log.d("EditProfile_ClickBtn", "Cancel Clicked") },
+                    Modifier.testTag("dismissBtn")) {
+                      Text("Cancel")
+                    }
+              }
+            }
+      }
+    }
+  }
+}
+
+/*@Preview(showBackground = true, widthDp = 540, heightDp = 1110)
+@Composable
+fun EditProfileDialoguePreview() {
+  val address = android.location.Address(java.util.Locale.getDefault())
+  address.countryCode = "CH"
+  address.locality = "Lausanne"
+  address.postalCode = "1000"
+  address.countryName = "Switzerland"
+  address.setAddressLine(0, "Rue de la Gare 1")
+  EditProfileDialog(onDismiss = { *//*TODO*//* }, onSave = { *//*TODO*//* }, dataUser = DataUser("Mr.", "John", "Doe", "John.Dow@example.com","+41223456789",address,"dummyPic.png","dummyUUID0000"))
```

Comment:
Why is this code in comment? If it isn't supposed to be used, then it needs to be deleted.

Reply:
This is a preview used in Android Studio to make UI design easier and faster by being able to render the UI without launching the app on the emulator, however , there is no way to "test" this part so it is currently commented to avoid reducing line coverage